File: domain/src/main/java/com/ahn/domain/setting/usecase/SaveThemeModeUseCase.kt
package com.ahn.domain.setting.usecase

import com.ahn.domain.setting.model.ThemeMode
import com.ahn.domain.setting.repository.ThemeRepository
import javax.inject.Inject

class SaveThemeModeUseCase @Inject constructor(private val repository: ThemeRepository) {
    suspend operator fun invoke(themeMode: ThemeMode) {
        repository.saveThemeMode(themeMode)
    }
}

File: ios/Packages/Domain/Sources/Domain/Setting/UseCase/SaveThemeModeUseCase.swift
public struct SaveThemeModeUseCase: Sendable {
    private let repository: any ThemeRepository

    public init(repository: any ThemeRepository) {
        self.repository = repository
    }

    public func callAsFunction(_ themeMode: ThemeMode) async throws {
        try await repository.saveThemeMode(themeMode)
    }
}
